File: Java/Vko2/Harjoittelija.java
import java.security.GeneralSecurityException;

public class Harjoittelija extends Opiskelija {

    private String mTyopaikka;
    private float mPalkka;

    public Harjoittelija() {
        mTyopaikka = "Tuyntematon";
        mPalkka = 0;

    }
     //Javassa kaikki tapahtuu rungossa, ei ole kaksoispisteitä niinkuin cpp:ssä
    public Harjoittelija(String aNimi, String aOpiskelijanumero, String aTyopaikka, float aPalkka) {
        super (aNimi, aOpiskelijanumero);
        mTyopaikka = aTyopaikka;
        mPalkka = aPalkka;
        System.out.println("Harjoittelijan 4 parametrinen rakentaja");
    }

    // Todo: setterit ja getterit harjoittelijan tiedoilla.
    public void tulostaTiedot() {
        System.out.println("Nimi: " + getNimi());
        System.out.println("Opintopisteet: " + getOpintopisteet());
        System.out.println("Palkka: " + mPalkka); //Tässäkin olisi hyvä käyttää getteriä
    }
}

File: C++/Vko1-2/Harjoittelija.cpp
#include "Harjoittelija.h"


Harjoittelija::Harjoittelija() : opiskelija(), mTyopaikka("tuntematon"), mPalkka(0)
{
}

Harjoittelija::Harjoittelija(const string& aNimi, const string& aOpiskelijanumero, const string& aTyopaikka, float aPalkka)
    : opiskelija(aNimi, aOpiskelijanumero), mTyopaikka(aTyopaikka), mPalkka(aPalkka) 
{
    cout << "Harjoittelijan rakentaja " << endl;
}

    //TODO GETTERIT JA SETTERIT


//Tulosta tiedot
void Harjoittelija::tulostaTiedot() const {

    //Voidaan tulostaa myös ensin kantaluokan tulostus esim. "opiskelija::tulostatiedot()"
    cout << "Nimi: " << getNimi() << endl;
    cout << "Opintopisteet: " << getOpintopisteet() << endl;
    cout << "Harjoittelupaikka: " << mTyopaikka << endl;
    cout << "Palkka: " << mPalkka << endl;

}
